### PR TITLE
forms mobile views

### LINKED ISSUE #113: Forms Mobile Views
Please create the mobile views for all forms - Add Box Form, Relocate Box Form, Pickup Box Form

### PR DESCRIPTION
**Pull request by ctc-devops**
Please create the mobile views for all forms - Add Box Form, Relocate Box Form, Pickup Box Form
closes #113

### CODE REVIEW

**Comment by meghakak** [comment]
We needed to create separate components for mobile and desktop for each of the forms because the fields are in different order in the mobile view vs desktop view. We kept the shared functions in the main file for each form (e.g. AddBoxForm.jsx), but everything related to yup is needed to be put in each component (both mobile and desktop). Let us know if y'all have a more efficient solution for this, but this is all we could think of to make the mobile view have the same order of fields as the figma.

**Comment by claudeisakeeb** [comment]
> We needed to create separate components for mobile and desktop for each of the forms because the fields are in different order in the mobile view vs desktop view. We kept the shared functions in the main file for each form (e.g. AddBoxForm.jsx), but everything related to yup is needed to be put in each component (both mobile and desktop). Let us know if y'all have a more efficient solution for this, but this is all we could think of to make the mobile view have the same order of fields as the figma.

UHHH I feel like the solution is to tell design and have them in the same order? Is there a reason that the orders are different? Just from a user's perspective I think it's confusing if the fields switch places for presumably no reason.

**Comment by meghakak** [comment]
> > We needed to create separate components for mobile and desktop for each of the forms because the fields are in different order in the mobile view vs desktop view. We kept the shared functions in the main file for each form (e.g. AddBoxForm.jsx), but everything related to yup is needed to be put in each component (both mobile and desktop). Let us know if y'all have a more efficient solution for this, but this is all we could think of to make the mobile view have the same order of fields as the figma.
> 
> UHHH I feel like the solution is to tell design and have them in the same order? Is there a reason that the orders are different? Just from a user's perspective I think it's confusing if the fields switch places for presumably no reason.

True, will ask design

**Comment by meghakak** [comment]
> > > We needed to create separate components for mobile and desktop for each of the forms because the fields are in different order in the mobile view vs desktop view. We kept the shared functions in the main file for each form (e.g. AddBoxForm.jsx), but everything related to yup is needed to be put in each component (both mobile and desktop). Let us know if y'all have a more efficient solution for this, but this is all we could think of to make the mobile view have the same order of fields as the figma.
> > 
> > 
> > UHHH I feel like the solution is to tell design and have them in the same order? Is there a reason that the orders are different? Just from a user's perspective I think it's confusing if the fields switch places for presumably no reason.
> 
> True, will ask design

The forms are now refactored such that Pickup and Relocate box form use one component for both desktop and mobile views. The add box form will remain two separate components because the ordering of fields must be different for better user experience.

File: src/components/AddBoxForm/AddBoxFormMobile.jsx
Comment (claudeisakeeb) [inline src/components/AddBoxForm/AddBoxFormMobile.jsx]: Can we add padding to the launched organically error message and make it line up with the Yes circle (like on the Figma)? 
image

File: src/components/AddBoxForm/AddBoxFormMobile.jsx
Comment (claudeisakeeb) [inline src/components/AddBoxForm/AddBoxFormMobile.jsx]: This is probably inconsequential but any error messages will disappear when you switch from mobile to desktop or vice versa (ie submit an empty form --> switch from mobile to desktop or vice versa --> error messages will disappear) 

I'm pretty sure you can do this with a useState in AddBoxForm but if it's not that simple I think it's fine.

File: src/components/BoxFormConfirmation/BoxFormConfirmation.jsx
Comment (claudeisakeeb) [inline src/components/BoxFormConfirmation/BoxFormConfirmation.jsx:24]: The "Back to Home" button is all the way at the bottom of the page on the mobile version and I have to scroll down to see it. Can we make the button and message fit in one screen? Also it should be colored like in the Figma (right now it's teal when it should be dark blue)

**Review by claudeisakeeb — CHANGES_REQUESTED** [review]
Mobile forms look great, just a few comments :D

File: src/components/AddBoxForm/AddBoxFormMobile.jsx
Comment (meghakak) [reply src/components/AddBoxForm/AddBoxFormMobile.jsx]: All the yup-related variables need to be instantiated in the component its being used in from what I've seen so it wouldn't work to have the errors variable in a useState in AddBoxForm since it's coming from yup. Also, in general, we don't need to worry too much about any changes when switching between mobile and desktop views since that wouldn't happen unless the user is drastically changing screen sizes.

File: src/components/AddBoxForm/AddBoxFormMobile.jsx
Comment (meghakak) [reply src/components/AddBoxForm/AddBoxFormMobile.jsx]: Nice catch, fixed!

File: src/components/BoxFormConfirmation/BoxFormConfirmation.jsx
Comment (meghakak) [reply src/components/BoxFormConfirmation/BoxFormConfirmation.jsx:24]: Fixed the styling on all the mobile view options I have and it looks fine. Let me know if this styling is still off for you. Regarding the colors, I mentioned this in DMs but a lot of buttons are currently using teal - we will change all of these at once when the style guide is ready.

File: src/components/AddBoxForm/AddBoxFormMobile.jsx
Comment (sylvesteryue) [reply src/components/AddBoxForm/AddBoxFormMobile.jsx]: Besides the errors, I think the re-rendering of the form from desktop to mobile also clears whatever the user had typed in the boxes beforehand, so we can properly save the form data using useState or use defaultValues from useForm()? I am not sure if that would be important, but I want to point that out before we approve, but if it doesn't matter we can just close and merge :)

**Review by sylvesteryue — CHANGES_REQUESTED** [review]
The mobile forms look great! I only have a question about the Add Box form, but other than that everything else :chefs_kiss:

File: src/components/AddBoxForm/AddBoxFormMobile.jsx
Comment (meghakak) [reply src/components/AddBoxForm/AddBoxFormMobile.jsx]: Refactored add box form so it is just one component now instead of two separate components for mobile and desktop views. This should fix this issue, but please test on your end as well

**Review by claudeisakeeb — APPROVED** [review]
cut the check

**Review by sylvesteryue — APPROVED** [review]
aight this looks pog

**Pull request merged**
